clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/Reindexer.java

```diff
@@ -110,20 +107,20 @@ public void reindex() throws ReindexingLockException {
     @SuppressWarnings("fallthrough") // (ノಠ ∩ಠ)ノ彡( \o°o)\
     private void progress(DocumentType type) {
         // If this is a new document type (or a new cluster), no reindexing is required.
-        reindexing = database.readReindexing();
-        status = reindexing.status().getOrDefault(type,
-                                                  Status.ready(clock.instant())
-                                                        .running()
-                                                        .successful(clock.instant()));
-        if (ready.get(type).isAfter(status.startedAt()))
-            status = Status.ready(clock.instant()); // Need to restart, as a newer reindexing is required.
-
-        database.writeReindexing(reindexing = reindexing.with(type, status));
-        metrics.dump(reindexing);
-
-        switch (status.state()) {
+        AtomicReference<Reindexing> reindexing = new AtomicReference<>(database.readReindexing());
```

Comment:
You should probably use a shared monitor/lock for both `reindexing` and `status` (unless they are used completely  independently of each other).

Reply:
They are used independently enough imo.